server/src/main/java/org/elasticsearch/cluster/coordination/PublicationTransportHandler.java

```diff
@@ -317,8 +316,8 @@ public class PublicationContext extends AbstractRefCounted {
         private final boolean sendFullVersion;
 
         // All the values of these maps have one ref for the context (while it's open) and one for each in-flight message.
-        private final Map<Version, ReleasableBytesReference> serializedStates = new ConcurrentHashMap<>();
-        private final Map<Version, ReleasableBytesReference> serializedDiffs = new HashMap<>();
+        private final Map<TransportVersion, ReleasableBytesReference> serializedStates = new ConcurrentHashMap<>();
+        private final Map<TransportVersion, ReleasableBytesReference> serializedDiffs = new HashMap<>();
```

Comment:
Although this represents a serialisation version, all uses of the map are actually from a `Version`

Reply:
this probably leads us back to the question we were wondering previously.
should a node also contain a transport version?

Reply:
In various places we need to know the transport version with which we should communicate with the remote node somehow (before we start to send the request, maybe because we must use entirely different actions on different versions). That doesn't mean it has to be on the `DiscoveryNode` tho, I think we could also do this by capturing the `Transport.Connection` earlier in the process and making decisions using its `getTransportVersion()` method instead.

Reply:
I think that is a separate piece of work, I'll make sure its recorded.